dcos/dcos-ui
# DCOS-38944 & DCOS-39074: add styling and dropdown truncation to header bar

* Adds the styling for the header bar elements: logo, side bar toggle, dropdowns
* Dropdowns are truncated and show ellipsis when the viewport is small
* Removes header bar from ui-kit-stage
* Enable the header bar

Closes DCOS-38944 & DCOS-39074

## Testing
* Checkout the branch on private referring to DCOS-38944
* Resize the screen and verify the dropdowns truncate
* The sidebar header is removed


## Result (For Truncation)
![header-bar-recording](https://user-images.githubusercontent.com/4956107/42981348-8a4e8366-8b90-11e8-8c8e-dbbb97201e89.gif)

## Trade-offs
* Keep the full logo as long as possible
* When the max-width is too small to show any meaningful text on the dropdown (chosen to be 100px), collapse the logo and expand the size of the dropdown
* Small intervals so there is not too much space between the logo and dropdowns

## Dependencies
Hold merge until the following are merged:
~~DCOS-39457 (https://github.com/dcos/dcos-ui/pull/3086)~~
~~DCOS-38942~~
~DCOS-39679~
